ffmpeg/ffmpeg_errors.go

```diff
@@ -49,6 +49,34 @@ func error_map() map[int]error {
 
 var ErrorMap = error_map()
 
+func non_retryable_errs() []string {
+	errs := []string{}
+	// Add in Cgo LPMS specific errors
+	for _, v := range lpmsErrors {
+		errs = append(errs, v.Desc)
+	}
+	// Add in internal FFmpeg errors
+	// from https://ffmpeg.org/doxygen/trunk/error_8c_source.html#l00034
+	ffmpegErrors := []string{
+		"Decoder not found", "Demuxer not found", "Encoder not found",
+		"Muxer not found", "Option not found", "Invalid argument",
+	}
+	for _, v := range ffmpegErrors {
+		errs = append(errs, v)
+	}
+	// Add in ffmpeg.go transcoder specific errors
+	transcoderErrors := []error{
+		ErrTranscoderRes, ErrTranscoderInp, ErrTranscoderVid, ErrTranscoderFmt,
```

Comment:
I think "ErrTranscoderInp" is OOM variable, what do you think of this?

Reply:
I see that after this PR the `ErrTranscoderInp` error is only used for the case when the function is called with `nil` input. https://github.com/livepeer/lpms/blob/560b4f72440a16cc08464608f27767bb9f171b32/ffmpeg/ffmpeg.go#L185

Can you elaborate a bit on what is meant by OOM here?

Reply:
> I see that after this PR the ErrTranscoderInp error is only used for the case when the function is called with nil input. 

In any case, this error will no longer be triggered by a 'bad video' segment, and only in case somebody uses the `ffmpeg.Transcode(` api with nil input. Will remove it from this list in an update!